internal/cmd/operator-sdk/generate/internal/relatedimages.go

```diff
@@ -83,6 +85,28 @@ func (c *relatedImageCollector) collectFromEnvironment(containerRef string, env
 	return nil
 }
 
+func (c *relatedImageCollector) collectFromBaseCSV(base *operatorsv1alpha1.ClusterServiceVersion) error {
+	if base == nil || base.Spec.RelatedImages == nil {
+		return nil
+	}
+
+	for _, relatedImage := range base.Spec.RelatedImages {
+		c.collect(relatedImage.Name, relatedImage.Image, "baseCSV")
+		if isStaticImage(relatedImage.Image) {
+			log.Warnf(
+				"warning: relatedImage %s is defined statically in base CSV as %s. Persistence of static related images will be removed in OperatorSDK v2",
+				relatedImage.Name, relatedImage.Image,
+			)
+		}
+	}
+
+	return nil
+}
+
+func isStaticImage(image string) bool {
+	return !strings.Contains(image, "@sha256:")
+}
```

Comment:
This is out of the responsibility of this package. All handling of pinned images is done in [this library][https://github.com/operator-framework/operator-manifest-tools/tree/main/pkg/image].